syncer/syncer.go

```diff
@@ -1895,6 +1945,10 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 
 			job := newXIDJob(currentLocation, startLocation, currentLocation)
 			err2 = s.addJobFunc(job)
+			if s.waitXIDJob == waitComplete {
+				runCancel()
```

Comment:
Should not call this cancel, it will cause other sub routine like syncDML or syncDDL exit before finish all remain tasks.

Reply:
I think we can retain it. Create `dmlCtxs` and `ddlCtx` depend on `runCtx` in syncDML and syncDDL. When the `runCtx` canceled, `dmlCtxs` and `ddlCtx` will cancel until theri job completed.

Reply:
Simple remove this line and depend on the defer block to do similar things is enough.

Reply:
> Simply remove this line and depend on the defer block to do similar things is enough.

Is it also exit above problem?

> Should not call this cancel, it will cause other sub routine like syncDML or syncDDL exit before finish all remain tasks.